### PR TITLE
Fixed issue with USB redirection

### PR DESCRIPTION
**Pull request by uplusplus**
Fixed issue with USB redirection not properly removing the device

### CODE REVIEW

**Comment by akallabeth** [comment]
@uplusplus What path did you fix with these changes? Looks like you fixed one and break others if I'm not missing something.

File: channels/drdynvc/client/drdynvc_main.c [hunk 1]
```diff
@@ -409,11 +409,25 @@ static UINT dvcman_write_channel(IWTSVirtualChannel* pChannel, ULONG cbSize,
 static UINT dvcman_close_channel_iface(IWTSVirtualChannel* pChannel)
 {
 	DVCMAN_CHANNEL* channel = (DVCMAN_CHANNEL*) pChannel;
+	DrdynvcClientContext* context;
+	UINT error = CHANNEL_RC_OK;
+	WLog_INFO(TAG, "close_channel_iface: id=%"PRIu32"", channel->channel_id);
 
-	if (!channel)
-		return CHANNEL_RC_BAD_CHANNEL;
+	if (channel->status == CHANNEL_RC_OK)
```
Comment (uplusplus) [inline channels/drdynvc/client/drdynvc_main.c]: Refer: urbdrc_search_usb_device
When usb device was removed, channel->Close() was closed to close the channel for this device.
As the channel->Close reference to dvcman_close_channel_iface.
According to the RDPEUSB document, we should remove the channel, So I copied some lines from dvcman_close_channel to dvcman_close_channel_iface;

I missed the NULL check - -||

Comment (uplusplus) [inline channels/drdynvc/client/drdynvc_main.c]: right

**Comment by uplusplus** [comment]
You may need to test the USB redirection module to confirm the cause of this fix

**Comment by akallabeth** [comment]
@uplusplus I get your reasoning, I'm just a bit cautious here. Specially since the USB module has not been updated in a long time and the rest of the project has improved quite a bit.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by AzureusNation** [comment]
@akallabeth couldn't this be closed now as your rewrite PR got merged?

**Comment by akallabeth** [comment]
superseeded by #5745

**Pull request closed without merging**